ui/src/components/paste_files_with_shortcut/mod.rs

```diff
@@ -8,6 +12,26 @@ use dioxus_hooks::{to_owned, use_ref};
 
 use crate::utils::clipboard::clipboard_data::get_files_path_from_clipboard;
 
+lazy_static! {
+    static ref LAST_CALLED: Mutex<Instant> = Mutex::new(Instant::now() - Duration::from_secs(2));
+}
+
+fn debounced_callback<F: FnOnce()>(callback: F, debounce_duration: Duration) {
+    let mut last_called = match LAST_CALLED.lock() {
```

Comment:
you can unwrap instead of using a match condition since its unlikely for the thread to panic while this lock is being held.